crates/ruff_workspace/src/cli.rs

```diff
@@ -0,0 +1,157 @@
+                .chain(std::iter::once(name))
+                .collect::<Vec<_>>()
+                .join(".");
+
+            self.values.push((fqn, field.doc.to_owned()));
+        }
+    }
+
+    /// Opaque type for option strings on the command line.
+    #[derive(Clone, Debug)]
+    pub struct OptionString(String);
+
+    impl From<String> for OptionString {
+        fn from(s: String) -> Self {
+            OptionString(s)
+        }
+    }
+
+    impl From<OptionString> for String {
+        fn from(o: OptionString) -> Self {
+            o.0
+        }
+    }
+
+    impl From<&str> for OptionString {
+        fn from(s: &str) -> Self {
+            OptionString(s.to_string())
+        }
+    }
+
+    impl std::ops::Deref for OptionString {
+        type Target = str;
+
+        fn deref(&self) -> &Self::Target {
+            &self.0
+        }
+    }
+
+    impl FromStr for OptionString {
+        type Err = ();
+
+        fn from_str(s: &str) -> Result<Self, Self::Err> {
+            Options::metadata()
+                .has(s)
+                .then(|| OptionString(s.to_owned()))
+                .ok_or(())
+        }
+    }
+
+    #[derive(Clone)]
+    pub struct OptionStringParser;
+
+    impl ValueParserFactory for OptionString {
+        type Parser = OptionStringParser;
+
+        fn value_parser() -> Self::Parser {
+            OptionStringParser
+        }
+    }
+
+    impl TypedValueParser for OptionStringParser {
+        type Value = OptionString;
+
+        fn parse_ref(
+            &self,
+            cmd: &clap::Command,
+            arg: Option<&clap::Arg>,
+            value: &std::ffi::OsStr,
+        ) -> Result<Self::Value, clap::Error> {
+            let value = value
+                .to_str()
+                .ok_or_else(|| clap::Error::new(clap::error::ErrorKind::InvalidUtf8))?;
+
+            value.parse().map_err(|()| {
+                let mut error =
+                    clap::Error::new(clap::error::ErrorKind::ValueValidation).with_cmd(cmd);
+                if let Some(arg) = arg {
+                    error.insert(
+                        clap::error::ContextKind::InvalidArg,
+                        clap::error::ContextValue::String(arg.to_string()),
+                    );
+                }
+                error.insert(
+                    clap::error::ContextKind::InvalidValue,
+                    clap::error::ContextValue::String(value.to_string()),
+                );
+                error
+            })
+        }
+
+        fn possible_values(&self) -> Option<Box<dyn Iterator<Item = PossibleValue> + '_>> {
+            let mut visitor = CollectOptionsVisitor::default();
+            Options::metadata().record(&mut visitor);
+
+            Some(Box::new(visitor.into_iter().map(|(name, doc)| {
+                let first_paragraph = doc
+                    .lines()
+                    .take_while(|l| !l.trim_end().is_empty())
+                    .map(|s| s.replace('"', "'"))
```

Comment:
Can you explain why replacing all `"` with `'` is necessary? It does seem dangerous to me in the case the doc contains any code examples

Reply:
It would be great if you could add an inline comment that summarises what you wrote in this comment. It will help future readers to understand (without guessing) of what's happening here.